public/app/features/dashboard/components/GenAI/hooks.ts

```diff
@@ -0,0 +1,97 @@
+import { useState } from 'react';
+import { useAsync } from 'react-use';
+import { Subscription } from 'rxjs';
+
+import { llms } from '@grafana/experimental';
+
+// Declared instead of imported from utils to make this hook modular
+// Ideally we will want to move the hook itself to a different scope later.
+type Message = llms.openai.Message;
+
+// TODO: Add tests
+export function useOpenAIStream(
+  model = 'gpt-3.5-turbo',
+  temperature = 1
+): {
+  setMessages: React.Dispatch<React.SetStateAction<Message[]>>;
+  reply: string;
+  inProgress: boolean;
+  loading: boolean;
```

Comment:
From the API perspective, I feel like having a `loading` and `inProgress` boolean is confusing. Should we just unify them under `loading` or `inProgress`?

Reply:
Loading has to be a separate variable because it's part of the useAsync state, not the actual API call state.